apps/ui/mix.exs

```diff
@@ -0,0 +1,47 @@
+defmodule Ui.Mixfile do
+  use Mix.Project
+
+  def project do
+    [
+      app: :ui,
+      version: "0.0.1",
+      build_path: "../../_build",
+      config_path: "../../config/config.exs",
+      deps_path: "../../deps",
+      lockfile: "../../mix.lock",
+      elixir: "~> 1.4",
+      elixirc_paths: elixirc_paths(Mix.env),
+      compilers: [:phoenix, :gettext] ++ Mix.compilers,
+      start_permanent: Mix.env == :prod,
+      deps: deps()
+    ]
+  end
+
+  # Configuration for the OTP application.
+  #
+  # Type `mix help compile.app` for more information.
+  def application do
+    [
+      mod: {Ui.Application, []},
+      extra_applications: [:logger, :runtime_tools, :backend]
+    ]
+  end
+
+  # Specifies which paths to compile per environment.
+  defp elixirc_paths(:test), do: ["lib", "test/support"]
+  defp elixirc_paths(_),     do: ["lib"]
+
+  # Specifies your project dependencies.
+  #
+  # Type `mix help deps` for examples and options.
+  defp deps do
+    [
+      {:phoenix, "~> 1.3.0"},
+      {:phoenix_pubsub, "~> 1.0"},
+      {:phoenix_html, "~> 2.10"},
+      {:phoenix_live_reload, "~> 1.0", only: :dev},
+      {:gettext, "~> 0.11"},
+      {:cowboy, "~> 1.0"}
+    ]
```

Comment:
there should be dependency on backend here. something like
{:backend, in_umbrella: true}

Reply:
@hubertlepicki I added :backend dependency to `extra_applications` a few lines earlier. Is it not enough?

Reply:
read this: 
https://www.amberbit.com/blog/2017/9/22/elixir-applications-vs-extra_applications-guide/